gpslogger/src/main/java/com/mendhak/gpslogger/common/Systems.java

```diff
@@ -234,4 +240,30 @@ else if(appThemeSetting.equalsIgnoreCase("light")){
 
     }
 
+    /**
+     * Returns a OneTimeWorkRequest with the given worker class and data map. The constraints are set to
+     * UNMETERED network type if the user has set the app to only send on wifi. Otherwise it is set to
+     * CONNECTED. The initial delay is set to 1 second to avoid the work being enqueued immediately.
+     * The backoff criteria is set to exponential with a 30 second initial delay.
+     * @param workerClass
+     * @param dataMap
+     * @return
+     */
+    public static OneTimeWorkRequest getBasicOneTimeWorkRequest(Class workerClass, HashMap<String, Object> dataMap) {
+
+        androidx.work.Data data = new Data.Builder().putAll(dataMap).build();
+
+        Constraints constraints = new Constraints.Builder()
+                .setRequiredNetworkType(PreferenceHelper.getInstance().shouldAutoSendOnWifiOnly() ? NetworkType.UNMETERED: NetworkType.CONNECTED)
+                .build();
+
+        OneTimeWorkRequest workRequest = new OneTimeWorkRequest
+                .Builder(workerClass)
+                .setConstraints(constraints)
+                .setInitialDelay(1, java.util.concurrent.TimeUnit.SECONDS)
+                .setBackoffCriteria(BackoffPolicy.EXPONENTIAL, 30, java.util.concurrent.TimeUnit.SECONDS)
```

Comment:
Might need to make the backoff policy to be linear if the user is interactively uploading a file.

Reply:
Tried this and it wasn't very useful — the backoff increases over time no matter what; the only difference is linear vs exponential and it isn't that different in terms of time taken. No point, so I'll just keep it the same and stay exponential.